.github/workflows/notebook-bot.yml

```diff
@@ -4,24 +4,28 @@ on:
   pull_request_target:
     types: [opened, reopened]
 
-permissions:
-  contents: read
+# Restrict permissions by default
+permissions: {}
 
 jobs:
   comment-welcome:
     permissions:
       contents: read
-      pull-requests: write
+      pull-requests: write  # Required for commenting on PRs
 
     runs-on: ubuntu-latest
+    timeout-minutes: 10
+
     steps:
     - name: Fetch pull request branch
-      uses: actions/checkout@v4
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
       with:
         repository: ${{ github.event.pull_request.head.repo.full_name }}
         ref: ${{ github.event.pull_request.head.sha }}
+
     - name: Fetch base develop branch
       run: git fetch -u "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" develop:develop
+
     - name: Create message
```

Comment:
❗ @bigbitbus, @alexnorell, Let's take a different approach and remove this action entirely (none of the steps are required).

* The comment informing about ReviewNB is written by reviewnb-bot ([example](https://github.com/roboflow/supervision/pull/645#issuecomment-2509554234))
* We were using this action temporarily, before we got the ReviewNB permissions approved. https://github.com/roboflow/supervision/pull/1588#issuecomment-2405844414